caterva2/services/sub.py

```diff
@@ -344,44 +345,34 @@ async def partial_download(abspath, path, slice_):
                 await download_chunk(path, schunk, n)
 
 
-@app.get('/api/download/{path:path}')
-async def download_data(path: str, slice_: str = None, download: bool = False):
+@app.get('/api/fetch/{path:path}')
+async def fetch_data(path: str, slice_: str = None, as_schunk: bool = False):
     """
-    Download or fetch a dataset.
+    Fetch a dataset.
 
     Parameters
     ----------
     path : str
         The path to the dataset.
     slice_ : str
         The slice to fetch.
-    download : bool
-        True if the intent is to download the dataset from 'files/'.  If False, the data is
-        returned as a StreamingResponse (it is 'fetched').
+    as_schunk : bool
+        True if the client accepts Blosc2 schunks.
```

Comment:
Since we're implementing a REST interface, this should ideally be an `Accept:` header sent by the client. Unfortunately there are no registered content (MIME) types for Python pickles nor Blosc2 schunks, so the query argument should do. :grin:

However, `as_schunk` is little more that a hint to the subscriber on the preferred format of the response. `prefer_schunk` may be more precise, but it's also longer.

Reply:
Yeah, I'm all in favor of implementing an `Accept` header for this sort of things; we need to discuss this further.

Also, +1 on `as_schunk` -> `prefer_schunk`.